astropy/time/core.py

```diff
@@ -2276,6 +2284,72 @@ def __add__(self, other):
     def __radd__(self, other):
         return self.__add__(other)
 
+    def mean(
+            self: 'Time',
+            axis: Optional[Union[int, Tuple[int]]] = None,
+            dtype: 'npt.DTypeLike' = None,
+            out: Optional['Time'] = None,
+            keepdims: bool = False,
+            *,
+            where: 'npt.NDArray[bool]' = True,
+    ) -> 'Time':
+        """Mean along a given axis.
+
+        This is similar to :meth:`~numpy.ndarray.mean`, but adapted to ensure
+        that the full precision given by the two doubles ``jd1`` and ``jd2``
+        is used, and that corresponding attributes are copied.
+
+        Note that the ``out`` argument is present only for compatibility with
+        ``np.mean``; since `Time` instances are immutable, it is not possible
+        to have an actual ``out`` to store the result in.
+
+        Similarly, the ``dtype`` argument is also present for compatibility with :meth:`~numpy.ndarray.mean`
+        and the ``dtype`` is determined by the ``Time`` format. An :exc:`~ValueError` is raised if this is set to anything
+        other than `None`.
+
+        Note that this implementation may lead to possible loss of precision for instances of ``Time`` spanning many
+        years. Users needing an accurate mean should look to :meth:`~math.fsum` instead.
+        """
+        if dtype is not None:
+            raise ValueError('Cannot set ``dtype`` on `Time` instances')
+        if out is not None:
+            raise ValueError("Since `Time` instances are immutable, ``out`` "
+                             "cannot be set to anything but ``None``.")
+        if self.masked:
+            if where is not None:
+                raise ValueError("The `where` keyword argument cannot be used with instances of MaskedArray")
+
+        kwargs_mean = dict(
+            axis=axis,
+            keepdims=keepdims,
+            where=where,
+        )
+        if axis is None:
+            divisor = self.size
+        else:
+            if isinstance(axis, int):
+                divisor = self.shape[axis]
+            else:
+                divisor = np.array([self.shape[i] for i in axis]).prod()
```

Comment:
This will not be right if `where` is involved. I'd generally not recommend using private functions from other packages, but here perhaps `numpy.core._methods._count_reduce_items` is your friend...

Alternatively, maybe easier is to use what is done for `Masked` -- that would mean broadcasting `where` but no more worries about checking types of `axis`, etc.: https://github.com/astropy/astropy/blob/31c6b608ec26b27fe61146699f2ffe8775474b43/astropy/utils/masked/core.py#L1058-L1078

Reply:
After updating this to use `numpy.core._methods._count_reduce_items()` I am confronted with another problem: what should we do if the divisor is zero?

If I try and just set the value to NaN, I get the error: `TypeError: Input values for jd class must be finite doubles`. Is there any way to express NaN using the `Time` class, or do we need to throw a `ValueError` if a zero-valued divisor is detected?

Reply:
@byrdie - wow, immediately more problems! There is an issue about having `NaT` - #6509 - and even an attempt at an implementation - #10272 - but right now `Time` does not support it, so I would suggest raising a `ValueError`.

Reply:
I just discovered that `numpy.core._methods._count_reduce_items()` does not support `keepdims` for the minimum version of numpy, so I changed it to the second method you suggested.

Reply:
Oops, sorry about that! Odd, I thought `keepdims` had been around more than long enough. Anyway, I guess it shows the risk of trying to depend on private functions!